packages/next/build/babel/plugins/next-page-config.ts

```diff
@@ -84,12 +88,62 @@ export default function nextPageConfig({
                   state.bundleDropped = true
                   return
                 }
+
+                state.isPrerender = config.experimentalPrerender === true
               },
             },
             state
           )
         },
       },
+      ExportDefaultDeclaration(path, state: ConfigState) {
```

Comment:
This can be simplified to:
```suggestion
      ExportDefaultDeclaration(path, state: ConfigState) {
        if (!state.isPrerender) {
          return
        }

        const prev = t.cloneDeep(path.node.declaration)
        if (prev.type.endsWith('Declaration')) {
          prev.type = prev.type.replace(/Declaration$/, 'Expression') as any
        }

        path.insertBefore([
          t.variableDeclaration('const', [
            t.variableDeclarator(t.identifier(pageComponentVar), prev as any),
          ]),
          t.assignmentExpression(
            '=',
            t.memberExpression(
              t.identifier(pageComponentVar),
              t.identifier(prerenderId)
            ),
            t.booleanLiteral(true)
          ),
        ])
        path.node.declaration = t.identifier(pageComponentVar)
      },
```

Reply:
We should probably make sure changing the type from `Declaration` to `Expression` doesn't have any negative side affects in babel since it does throw an error without. If not, nice workaround 👍

Reply:
It's deep cloned and Babel will check the type anyway on insertion, AFAIK Declarations and Expressions are identical, just refer to their scope (behind a var or not). A fake type wouldn't work.